File: src/main/kotlin/ro/bogdannegoita/myplannerkt/domain/Task.kt
package ro.bogdannegoita.myplannerkt.domain

import ro.bogdannegoita.myplannerkt.commons.TaskDto

class Task(
    private val data: TaskDto,
) : Comparable<Task> {
    val id = data.id!!
    val title by data::title
    val description by data::description
    val index by data::index

    override fun compareTo(other: Task) = index.compareTo(other.index)
}

File: MyPlannerWeb/src/models/Task.js
export default class Task {
	id;
	title;
	description;
	index;

	constructor(task) {
		if (!task) return;
		this.id = task.id;
		this.title = task.title;
		this.description = task.description;
		this.index = task.index;
	}
}
